### DISCUSSION TITLE
Use Auth Bearer instead of Cookies

### DISCUSSION THREAD

**Discussion by HackedOS** [maxcountryman/axum-login#218] [Ideas]
Currently axum-login only supports cookies, but some might want to use auth bearer headers instead, so basically
one would have a endpoint at which they can authenticate with username/password/oauth/whatever which sends a response with a token. here onwards the client can set this token as the auth header.

This should be fairly easy to implement. Two ways to go about this 
- having two features one for cookie based another for header based
- having an `login_token()` function that gives the token

I think method two is better, should be fairly simple:
- have an `login_token()` function which generates the token, it can be the same as the auth cookie
- update the session middleware to check for both the cookie and the auth header

the auth header can be treated the exact same way as the cookie server-side, only client side it needs to be manually set and not automatically by the browser.

**Comment by maxcountryman** [comment]
> having an login_token() function that gives the token

Because your backend type is provided by you, you can already do this.

For example, if you implement your backend as a struct then you can simply implement a method `login_token` that generates a token according to whatever spec you like; in your handler, return this token.

Likewise you can also support alternative forms of authentication within the same handler or other handlers.

As mentioned [here](https://github.com/maxcountryman/axum-login/discussions/214), you would provide a middleware to consume the token instead of the session cookie.
